proto/spacemesh/v1/app_types.proto

```diff
@@ -0,0 +1,129 @@
+syntax = "proto3";
+package spacemesh.v1;
+option go_package = "github.com/spacemeshos/api/release/go/spacemesh/v1";
+import "spacemesh/v1/types.proto";
+
+// Spacemesh app primitive variable types
+enum VariableType {
+    VARIABLE_TYPE_UNSPECIFIED = 0;
+    VARIABLE_TYPE_I8 = 1;
+    VARIABLE_TYPE_U8 = 2;
+    VARIABLE_TYPE_I16 = 3;
+    VARIABLE_TYPE_U16 = 4;
+    VARIABLE_TYPE_I32 = 5;
+    VARIABLE_TYPE_U32 = 6;
+    VARIABLE_TYPE_I64 = 7;
+    VARIABLE_TYPE_U64 = 8;
+    VARIABLE_TYPE_BOOL = 9;
+    VARIABLE_TYPE_AMOUNT = 10;
+    VARIABLE_TYPE_ADDRESS = 11;
+    // future supported types will be added here
+}
+
+// Variable metadata
+message VariableMetadata {
+    uint32 id = 1; // var index in app's state
+    string name = 2; // var name
+    VariableType type = 3; // type
+    bool is_array = 4; // when true, variable is an array of 0 or more primitive value types, otherwise, a single primitive type value.
+}
+
+message ValuesArray {
+    repeated bytes values = 1;
+}
+
+// The meta-data and value of an app's variable
+message Variable {
+    VariableMetadata meta_data = 1;
+    oneof data {
+        bytes value = 2; // since bytes size is arbitrary - this will support future var types
+        ValuesArray values = 3;
+    }
+}
+
+// The address of an app template in the templates global state store
+message TemplateAddress {
+    bytes address = 1;
+}
+
+// Information about an app template in the global state store
+message Template {
+    TemplateAddress address = 1; // template address
+    string name = 2; // template name
+
+    // note - the author can be obtained from the deploy transaction but it is nice to have it here
+    AccountId author = 3; // template uploader
+
+    // The layer can be obtained from the tx receipt but it is convenient to have it here
+    LayerNumber deploy_layer = 4; // layer that the template was deployed - 0 for genesis templates
+    TransactionId deploy_tx_id = 5; // the id of the deploy template transaction. Empty for genesis templates.
+    bytes code = 6; // template wasm code
+    repeated VariableMetadata vars = 7; // template variables order and metadata
+}
+
+// A Spacemesh app
+// Note that both layer and spawner can be obtained from the transaction and transaction id for apps spawned by a transaction
+// but is convenient to return them directly as part of apps' metadata.
+message App {
+    AccountId address = 1; // App's address in global space
+    string name = 2; // short descriptive name
```

Comment:
Do we need this?

Reply:
I think it is good to have a short app name that describes it. I think that it is something that is set by spwaner and spawn time per app.